crates/node/builder/src/builder/mod.rs

```diff
@@ -295,12 +314,13 @@ where
     T: NodeTypes,
 {
     /// Advances the state of the node builder to the next state where all components are configured
-    pub fn with_components<CB>(
+    pub fn with_components<CB, AO>(
```

Comment:
can we document that this type also expects the addons types for additional functionality?